### PR TITLE
Add rule E3033 to validate the min and max of a string

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*
#739 
*Description of changes:*
- Add new rule E3033 to test the minimum and maximum length of a string

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/data/CloudSpecs/ap-northeast-1.json [hunk 1]
```diff
@@ -32213,10 +32213,7 @@
           "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-performanceinsightsretentionperiod",
           "PrimitiveType": "Integer",
           "Required": false,
-          "UpdateType": "Mutable",
-          "Value": {
-            "ValueType": "PerformanceInsightsRetentionPeriod"
-          }
+          "UpdateType": "Mutable"
```
Comment (fatbasstard) [inline src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: You're reverting other changes... Can you get your code up-to-date?

File: src/cfnlint/data/CloudSpecs/ap-northeast-1.json [hunk 2]
```diff
@@ -35816,6 +35813,11 @@
         "updateReference"
       ]
     },
+    "CodeCommitRepositoryName": {
```
Comment (fatbasstard) [inline src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: ❤️

**Comment by fatbasstard** [comment]
Can you merge in the SecurityGroup Description?
Then you can remove this rule:
https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/resources/ectwo/SecurityGroupDescription.py

🎉

**Comment by kddejong** [comment]
Now includes removal of the SG Description rule
